lxd/db/update.go

```diff
@@ -91,14 +97,72 @@ func UpdatesApplyAll(db *sql.DB, doBackup bool, postApply func(int) error) error
 			backup = true
 		}
 		logger.Debugf("Updating DB schema from %d to %d", version, version+1)
-		if postApply != nil {
-			return postApply(version + 1)
+		if hook != nil {
+			// XXX We need to commit the transaction before the
```

Comment:
Why the "XXX" here?

Reply:
Ah, so far I've only ever seen us use XXX for comments that are temporary and meant to go away before merging the branch.

Reply:
@stgraber sorry, that probably wasn't clear. The "XXX" is not temporary, it's permanent and warns the reader (who could be the initial reviewer or anybody else later in the lifetime of the code) that the implementation is sub-optimal. Happy to change/drop it if it's not a convention we want to have in LXD.

Reply:
What I meant is that so far the only times we've used "XXX" in LXD was for local comments while working on the code which must go away before merging.

If we expect that something is a current limitation and should be fixed at some later point (usually as part of a larger refactoring), we use FIXME.

Reply:
Fixed (replaced XXX with FIXME)